src/structures/OAuth2Application.js

```diff
@@ -103,6 +104,14 @@ class OAuth2Application {
        */
       this.owner = this.client.dataManager.newUser(data.owner);
     }
+
+    if (data.team) {
+      /**
+       * The owning team of this OAuth application
+       * @type {?Team}
+       */
+      this.team = new Team(this.client, data.team);
```

Comment:
The team property should always exist as null or Team, not excluding the property if no data.team, so the object keeps consistent shape.

Reply:
Well, and also, it should be deprecated since it will be gone in v12.

Reply:
How would such a deprecation look like? Emitting a process warning on access seems unjustified as there is no way to fix this before v12. Maybe a warn or info?

Reply:
At minimum just a docstring @deprecated. But probably that with an explanation (in a warn or info tag) in the description.